packages/iframe-stamper/src/index.ts

```diff
@@ -193,31 +243,13 @@ export class IframeStamper {
    */
   async injectCredentialBundle(bundle: string): Promise<boolean> {
     return new Promise((resolve, reject) => {
-      this.iframe.contentWindow?.postMessage(
-        {
-          type: IframeEventType.InjectCredentialBundle,
-          value: bundle,
-        },
-        "*"
-      );
-
-      window.addEventListener(
-        "message",
-        (event) => {
-          if (event.origin !== this.iframeOrigin) {
-            // There might be other things going on in the window, for example: react dev tools, other extensions, etc.
-            // Instead of erroring out we simply return. Not our event!
-            return;
-          }
-          if (event.data?.type === IframeEventType.BundleInjected) {
-            resolve(event.data["value"]);
-          }
-          if (event.data?.type === IframeEventType.Error) {
-            reject(event.data["value"]);
-          }
-        },
-        false
-      );
+      this.messageChannel.port1.postMessage({
+        type: IframeEventType.InjectCredentialBundle,
+        value: bundle,
+      });
+      this.messageChannel.port1.onmessage = (event) => {
+        this.onMessageHandler(event, resolve, reject);
+      };
```

Comment:
I don't think you need the wrapping promise here? How about modifying this function to be:

```ts
this.messageChannel.port1.postMessage({
    type: IframeEventType.InjectCredentialBundle,
    value: bundle,
});

return addMessageHandler();
```

(looks like you've done that for the other functions, `injectCredentialBundle` is the odd one out!)

Reply:
Talked through this on Slack: while odd indeed, we'd break the interface if we made this function synchronous, because it means callers would have to remove `await`s. So, it'd be a cleaner interface, but we're choosing to keep it the way it is for compatibility reasons. Let's not ask people to change their code, there's no real benefit here.

Reply:
Sorry if I'm missing something obvious — what is the difference in interface between the two? (`injectCredentialBundle` vs `injectKeyExportBundle`), i.e. why is `injectCredentialBundle` the odd one out?

Reply:
`injectCredentialBundle` returns a promise which wraps both the `postMessage` and `onmessage` logic:

```
    return new Promise((resolve, reject) => {
      this.messageChannel.port1.postMessage({
        type: IframeEventType.InjectCredentialBundle,
        value: bundle,
      });
      this.messageChannel.port1.onmessage = (event) => {
        this.onMessageHandler(event, resolve, reject);
      };
    });
```

`injectKeyExportBundle` calls `postMessage` and then returns a promise which handles the `onmessage` logic:

```
    this.messageChannel.port1.postMessage({
      type: IframeEventType.InjectKeyExportBundle,
      value: bundle,
      keyFormat,
      organizationId,
    });

    return this.addMessageHandler();
```